canvas_sdk/v1/data/base.py

```diff
@@ -91,3 +123,52 @@ def codings(value_set: Type["ValueSet"]) -> tuple[tuple[str, set[str]]]:
                 if i[0] in values_dict
             ),
         )
+
+
+class TimeframeLookupQuerySetProtocol(QuerySetProtocol):
+    """A typing protocol for use in TimeframeLookupQuerySetMixin."""
+
+    @property
+    @abstractmethod
+    def timeframe_filter_field(self) -> str:
+        """A protocol method for timeframe_filter_field."""
+
+    ...
```

Comment:
For abstract methods, I believe the correct thing to do is to raise a `NotImplementedError` rather than return `None`.

```suggestion
        raise NotImplementedError
    ...
```

Reply:
Added.